File: backend/app/schemas/merchant.py
"""
Purpose: Pydantic schemas for the merchant-listing API (plan.md Section 18).

Lets the storefront's landing page show every AI-transactable merchant
(currently UrbanNest and TechHub) without hardcoding their names/slugs on
the frontend.
"""
from pydantic import BaseModel


class MerchantResponse(BaseModel):
    """One merchant a buyer (or buyer agent) can shop at."""

    merchant_id: str
    slug: str
    name: str
    currency: str

File: frontend/src/types/merchant.ts
/**
 * TypeScript shape mirroring backend/app/schemas/merchant.py -- the
 * storefront's merchant picker (plan.md Section 18).
 */

/** One merchant a buyer (or buyer agent) can shop at. */
export interface MerchantResponse {
  merchant_id: string
  slug: string
  name: string
  currency: string
}
